### ISSUE TITLE
Set `snapshot_memory_limit` based on memory usage

### ISSUE THREAD

**Issue by vpetersson** [NebraLtd/hm-miner#74]
I order to support Raspberry Pis 4B in various configurations, we need to modify `snapshot_memory_limit` accordingly.

The easiest way to do this is probably to look for the total memory usage and set it to `TOTAL_MEMORY - 256MB` or something like that.

**Comment by shawaj** [comment]
@vpetersson any reason why we can't just use the default we are using in the raspi cm3 config?

**Comment by shawaj** [comment]
also - since we aren't taking snapshots on miners typically, this is probably irrelevant

**Comment by vpetersson** [comment]
Yeah this can be closed out. This was opened before they fixed the garbage collection issue. The default value should be fine.
